src/unittestScripts.ts

```diff
@@ -27,23 +27,22 @@ def get_tests(suite):
         return [suite]
 
 
-def discover_tests():
+def discover_with_loader():
     loader = TestLoader()
-    suites = loader.discover("${configuration.startDirectory}", pattern="${configuration.pattern}")
-    return get_tests(suites)
+    return loader.discover("${configuration.startDirectory}", pattern="${configuration.pattern}")
 
 
-def load_tests(test_names):
-    all_tests = discover_tests()
-    if not test_names:
-        return all_tests
-    return filter(lambda test: any(test.id().startswith(name) for name in test_names), all_tests)
+def discover_tests():
+    return get_tests(discover_with_loader())
 
 
 def run_tests(test_names):
-    tests = load_tests(test_names)
     runner = TextTestRunner(resultclass=TextTestResultWithSuccesses)
-    results = [runner.run(test) for test in tests]
+    if not test_names:
+        results = [runner.run(discover_with_loader())]
+    else:
+        results = [runner.run(loader.loadTestsFromName(name)) for name in test_names]
```

Comment:
Maybe I'm missing something, but isn't `loader` defined in this scope?

Reply:
Hmm, unsure how i didn't notice this. Will go fix.